submit-web/src/components/Submission/DocumentRow.tsx

```diff
@@ -17,27 +21,45 @@ export default function DocumentRow({
   documentSubmission,
   submissionItem,
 }: DocumentRowProps) {
+  const { submissionPackageId } = useParams({ strict: false });
   const [pendingGetObject, setPendingGetObject] = useState(false);
   const { setOpen: setOpenModal, setClose: setCloseModal } = useModal();
   const {
     submitted_document: { name, url },
     version,
     submitted_by,
   } = documentSubmission;
+  const isConsultationRecord =
+    name === SUBMISSION_ITEM_TYPE.CONSULTATION_RECORD;
+
+  const { mutate: updateStateSubmissionPackage } =
+    useUpdateStateSubmissionPackage({
+      onError: () => {
+        notify.error("Failed to update management plan");
+      },
+    });
 
   const openVerificationModal = () => {
     //todo: setup onConfirm
     setOpenModal(
       <ConfirmationModal
         onConfirm={() => {
           setCloseModal();
+          updateStateSubmissionPackage({
```

Comment:
currently with this:

when the modal pops up
- user clicks confirm
- modal immediately closes
- an api call happens in the backend

this way there's not error handling if the api call fails, and the user doesn't see anything loading

what we probably need:
- user clicks confirm
- confirm button loads while api call is pending
- after success the modal is closed